src/test/scala/com/cognite/sdk/scala/v1/WellDataLayerTest.scala

```diff
@@ -0,0 +1,194 @@
+          )
+        )
+      ),
+      (
+        "wellhead",
+        Json.fromFields(
+          Seq(
+            ("x", Json.fromDoubleOrNull(42.42)),
+            ("y", Json.fromDoubleOrNull(12.34)),
+            ("crs", Json.fromString("EPSG:4326"))
+          )
+        )
+      )
+    )
+
+    val expected = JsonObject(
+      ("matchingId", Json.fromString("deterministic-id")),
+      ("name", Json.fromString("my new well")),
+      (
+        "sources",
+        Json.arr(
+          Json.fromFields(
+            Seq(
+              ("sourceName", Json.fromString("A")),
+              ("assetExternalId", Json.fromString("well1"))
+            )
+          )
+        )
+      ),
+      ("wellbores", Json.arr()),
+      (
+        "wellhead",
+        Json.fromFields(
+          Seq(
+            ("x", Json.fromDoubleOrNull(42.42)),
+            ("y", Json.fromDoubleOrNull(12.34)),
+            ("crs", Json.fromString("EPSG:4326"))
+          )
+        )
+      )
+    )
+
+    client.wdl.setItems("wells", Seq(well1))
+    val results = client.wdl.listItemsWithPost("wells/list")
+    results.items.size shouldEqual 1
+
+    results.items.headOption.value.shouldDeepEqual(expected)
+  }
+
+  it should "create, retrieve, and delete wells" in {
+    // first, create a new source
+
+    val newSource = Source(name = shortRandom())
+    client.wdl.sources.create(Seq(newSource)).size shouldEqual 1
+
+    // Then set merge rules
+    client.wdl.wells.setMergeRules(WellMergeRules(Seq(newSource.name)))
+    client.wdl.wellbores.setMergeRules(WellboreMergeRules(Seq(newSource.name)))
+
+    client.wdl.wells.create(
+      Seq(
+        WellSource(
+          name = "my new well",
+          source = AssetSource("A:my new well", newSource.name),
+          wellhead = Some(Wellhead(0.0, 60.0, "EPSG:4326"))
+        )
+      )
+    ).size shouldEqual 1
+
+    client.wdl.wellbores.create(
+      Seq(
+        WellboreSource(
+          name = "my new wellbore",
+          source = AssetSource("A:my new wellbore", newSource.name),
+          datum = Some(Datum(54.9, "meter", "KB")),
+          wellAssetExternalId = "A:my new well"
+        )
+      )
+    ).size shouldEqual 1
+
+    val wells = client.wdl.wells.list()
+    wells.size shouldEqual 1
+    val well = wells.headOption.value
+    well.name shouldEqual "my new well"
+    well.sources.headOption.value should equal(AssetSource("A:my new well", newSource.name))
+    well.wellhead shouldEqual Wellhead(0.0, 60.0, "EPSG:4326")
+
+    well.wellbores match {
+      case None => sys.error("No wellbores")
+      case Some(wellbores) =>
+        wellbores.size shouldEqual 1
+        val wellbore = wellbores.headOption.value
+        wellbore.name shouldEqual "my new wellbore"
+        wellbore.sources shouldEqual Seq(AssetSource("A:my new wellbore", newSource.name))
+        wellbore.datum shouldEqual Some(Datum(54.9, "meter", "KB"))
+    }
+
+    client.wdl.sources.deleteRecursive(Seq(newSource))
```

Comment:
maybe adding a check with get request and make sure the data is deleted after this?

Reply:
done